File: src/lightnap-ng/src/app/core/backend-api/static-content-sort-by.ts
/*
 * Static content types used for sorting static contents.
 */
export type StaticContentSortBy = "Key" | "Status" | "Type" | "ReadAccess" | "CreatedDate" | "LastModifiedDate";

export const StaticContentSortBys = {
    Key: "Key",
    Status: "Status",
    Type: "Type",
    ReadAccess: "ReadAccess",
    CreatedDate: "CreatedDate",
    LastModifiedDate: "LastModifiedDate",
} as const;

File: src/LightNap.Core/StaticContents/Models/StaticContentSortBy.cs
namespace LightNap.Core.StaticContents.Models
{
    /// <summary>
    /// Supported static content sorting options.
    /// </summary>
    public enum StaticContentSortBy
    {
        /// <summary>
        /// Sort by the content key identifier.
        /// </summary>
        Key,

        /// <summary>
        /// Sort by the content status.
        /// </summary>
        Status,

        /// <summary>
        /// Sort by the content type.
        /// </summary>
        Type,

        /// <summary>
        /// Sort by the read access level.
        /// </summary>
        ReadAccess,

        /// <summary>
        /// Sort by the creation date.
        /// </summary>
        CreatedDate,

        /// <summary>
        /// Sort by the last modified date.
        /// </summary>
        LastModifiedDate
    }
}
